ext/bigdecimal/bigdecimal.c

```diff
@@ -2594,6 +2594,13 @@ static Real *BigDecimal_new(int argc, VALUE *argv);
  * ArgumentError:: If the +initial+ is a Float or Rational, and the +digits+
  *                 value is omitted, this exception is raised.
  */
+static VALUE
+BigDecimal_s_new(int argc, VALUE *argv, VALUE self)
+{
+  rb_warning("BigDecimal.new is deprecated");
```

Comment:
As a note, maybe this should have been `rb_warn`, as `rb_warning` only warns with `$VERBOSE=true` / `ruby -w`, not the default.

Reply:
bigdecimal 1.4.x always warns in `BigDecimal.new`.
See https://github.com/ruby/bigdecimal/blob/v1.4.x/lib/bigdecimal.rb#L10

Reply:
Right, but in e.g. MRI 2.5.3, bigdecimal 1.3.4 is default and so:
```
$ ruby -rbigdecimal -e 'p BigDecimal.new "2"'
0.2e1
```
does not warn (without -v/-w).